### PR TITLE
:bug: Fix wrong branch name display for weekly update script

### PR DESCRIPTION
**Pull request by chandankumar4**
**What this PR does / why we need it**:
Fix the branch name display issue after running the command `make release-weekly-update-tool` from branch other than `main`.

Example: Output of command after bug fix!!
```bash
$:: github.com/sigs.k8s.io/cluster-api ‹release-1.6*› » ./bin/weekly --from 2023-12-11 --to 2023-12-17 --milestone v1.7
Weekly update :rotating_light:
Changes from 2023-12-11 to 2023-12-17 a total of 7 new commits where merged into release-1.6.

- 1 :sparkles: New Features
         - KCP: Allow mutation of all fields that should be mutable (#9885)

- 4 :bug: Bug Fixes
         - capd: fix ignition to also set the kube-proxy configuration to skip setting sysctls (#9895)
         - Improve log k/v pairs and a improve/drop a few log lines (#9880)
         - ipam: fix webhooks using mixed api versions (#9863)
         - clusterctl: validate no objects exist from CRDs before deleting them (#9834)

- 2 Other changes :seedling:

All merged PRs can be viewed in GitHub:
https://github.com/kubernetes-sigs/cluster-api/pulls?q=is%3Apr+closed%3A2023-12-11..2023-12-18+is%3Amerged+milestone%3Av1.7+

_Thanks to all our contributors!_ 😊
/Your friendly comms release team

```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # N/A

/area release

### CODE REVIEW

File: hack/tools/release/weekly/main.go
Comment (willie-yao) [inline hack/tools/release/weekly/main.go:189]: I'm a bit confused what this block of code is doing. Does it set the branch to the last line of the branch name? I think it may be better to include the entire branch name if it spans multiple lines.

**Review by willie-yao — COMMENTED** [review]
Looks good pending one minor comment. Thanks for working on this!

File: hack/tools/release/weekly/main.go
Comment (chandankumar4) [reply hack/tools/release/weekly/main.go]: it was just removing any un-necessary lines if comes, but I think that is not required as this command will only return with current branch only. so without this it will also work fine.

**Review by willie-yao — APPROVED** [review]
/lgtm

**Comment by chandankumar4** [comment]
Hey @mcbenjemaa if you get time, can you please take a look once, I can see lgtm label is added but not merged yet. Thanks

**Comment by willie-yao** [comment]
This looks good to me. Just waiting on an approver to take a look. cc @sbueringer

File: hack/tools/release/weekly/main.go
Comment (cahillsf) [inline hack/tools/release/weekly/main.go:175]: this is a bit confusing, i think we can consolidate the `"Error"` string to print in the same line as `err`

File: hack/tools/release/weekly/main.go
Comment (cahillsf) [inline hack/tools/release/weekly/main.go:177]: is this a debug print statement?  if there is an error in attempting to assign the value and we are about to `return 1`, why print this?

File: hack/tools/release/weekly/main.go
Comment (chandankumar4) [reply hack/tools/release/weekly/main.go:175]: I have followed the same pattern which is there in line no: 105 after executing any command.

Comment (cahillsf) [reply hack/tools/release/weekly/main.go:175]: ah cool, if this is a pattern that works for the comms team it works for me 👍

File: hack/tools/release/weekly/main.go
Comment (chandankumar4) [reply hack/tools/release/weekly/main.go:177]: Similar comment as above.

**Comment by cahillsf** [comment]
question on your example output, is it expected to have the milestone minor version differ from the branch minor version?
`$:: github.com/sigs.k8s.io/cluster-api ‹release-1.6*› » ./bin/weekly --from 2023-12-11 --to 2023-12-17 --milestone v1.7`

the correspond docs don't really specify: https://github.com/kubernetes-sigs/cluster-api/blob/c8c07941da42cc90db170876e4e5bfd13cd533d9/docs/release/release-tasks.md?plain=1#L338

**Comment by chandankumar4** [comment]
> from the branch minor version

Yes, that's the expected behaviour, the use of `milestone minor version` is for targeting against which version the notes should be generated and `branch minor version` is indicating that from which branch notes is generated.
cc @willie-yao

**Comment by chandankumar4** [comment]
@cahillsf  In this link: https://kubernetes.slack.com/archives/C8TSNPY4T/p1704956238693459?thread_ts=1704956148.936569&cid=C8TSNPY4T milestone is `v1.7` and current branch is `release-1.5`

**Comment by chandankumar4** [comment]
Have checked with @willie-yao and he is good with this change!!

**Pull request merged**
